ngraph/python/src/ngraph/utils/tensor_iterator_types.py

```diff
@@ -0,0 +1,185 @@
+# ******************************************************************************
+# Copyright 2017-2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+"""Helper classes for aggregating TensorIterator input/output desciptor attributes."""
+
+from typing import List
+
+from ngraph.impl import Node
+from ngraph.impl.op import Parameter
+
+
+class GraphBody(object):
+    """Class containing graph parameters and results."""
+
+    def __init__(
+        self,
+        parameters: List[Parameter],
+        results: List[Node],
+    ) -> None:
+        self.parameters = parameters
+        self.results = results
+
+    def serialize(self) -> dict:
+        return {
+            "parameters": self.parameters,
+            "results": self.results,
+        }
+
+
+class TensorIteratorInputDesc(object):
+    """Represents a generic input descriptor for TensorIterator operator."""
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+    ) -> None:
+        self.input_idx = input_idx
+        self.body_parameter_idx = body_parameter_idx
+
+    def serialize(self) -> dict:
+        return {
+            "input_idx": self.input_idx,
+            "body_parameter_idx": self.body_parameter_idx,
+        }
+
+
+class TensorIteratorSliceInputDesc(TensorIteratorInputDesc):
+    """Represents a TI graph body input formed from slicec of an TI input."""
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+        start: int,
+        stride: int,
+        part_size: int,
+        end: int,
+        axis: int,
+    ) -> None:
+        super(TensorIteratorSliceInputDesc, self).__init__(input_idx, body_parameter_idx)
```

Comment:
Python 3 makes this simpler:
```suggestion
        super().__init__(input_idx, body_parameter_idx)
```